misc/package/Data Files/MWSE/core/lib/logger/init.lua

```diff
@@ -0,0 +1,653 @@
+    local modName, moduleName, modDir
+    if metadata then
+        local package = metadata.package ---@diagnostic disable-next-line: undefined-field
+        modName = package.shortName or package.shortName or package.name 
+    end
+
+    -- actual mod information starts at index 2 if there's an author name
+    local cutoff = hasAuthorName and 2 or 1
+
+    -- if the module name doesn't exist, use the mod folder name (excluding the author name)
+    modName = modName or luaParts[cutoff]
+    --[[ generate module name by combining everything together, ignoring the mod root.
+        examples: 
+            "herbert100/more quickloot/common.lua" ~> `moduleName = "common.lua"`
+            "herbert100/more quickloot/managers/organic.lua" ~> `moduleName = "managers/organic.lua"`
+    ]]
+    moduleName = table.concat(table.filterarray(luaParts, function(i) return i > cutoff end), "/")
+    if hasAuthorName then
+        -- e.g. modDir = "herbert100.more quickloot"
+        modDir = luaParts[1] .. "." .. luaParts[2]
+    else
+        -- e.g. modDir = "Expeditious Exit"
+        modDir = luaParts[1]
+    end
+    return modName, moduleName, modDir
+end
+
+
+---@class Logger.newParams
+---@field modName string? the name of the mod this logger is for
+---@field modDir string? the name of the mod this logger is for
+---@field level Logger.LEVEL|Logger.LEVEL_STRING|nil the log level to set this object to. Default: "INFO"
+---@field moduleName string? the module this logger belongs to, or `nil` if it's a general purpose logger
+---@field includeLineNumber boolean? should the current line be printed when writing log messages? Default: `true`
+---@field includeTimestamp boolean? should the current time be printed when writing log messages? Default: `false`
+---@field writeToFile string|boolean|nil whether to write the log messages to a file, or the name of the file to write to. if `false` or `nil`, messages will be written to `MWSE.log`
+
+
+--[[##Create a new logger for a mod with the specified `modName`. 
+
+New objects can be created in the following ways:
+- `log = Logger.new("MODNAME")`
+- `log = Logger.new{modName="MODNAME", level=LEVEL?, ...}`
+
+In addition to `modName`, you may specify
+- `level`: the logging level to start this logger at. This is the same as making a new logger and then calling `log:setLevel(level)`
+- `writeToFile`: either a boolean saying we should write to a file, or the name of a file to write to. If false (the default), then log messages will be written to `MWSE.log`
+]]
+---@param params Logger.newParams|string?
+---@return Logger
+function Logger.new(params, params2)
+    -- if it's just a string, treat it as the `modName`
+    if not params then
+        params = {}
+    elseif type(params) == "string" then
+        params = {modName=params} 
+
+    -- check if the user typed "Logger:new" instead of "Logger.new"
+    elseif params == Logger then
+        params = params2 or {}
+
```

Comment:
Let's leave it as it works currently: we accept only calling as `Logger.new`.

Reply:
sounds good. i'll revert this change